mdsuite/graph_modules/molecular_graph.py

```diff
@@ -247,14 +256,224 @@ def check_a_in_b(a, b):
+
+    def _adjacency_graph_isomorphism_test(self):
+        """
+        Determine approximate isomorphism between the computed adjacency graph and a
+        reference graph.
+
+        Returns
+        -------
+        Nothing, will raise an exception if the test fails.
+
+        Notes
+        -----
+        This must be implemented, however, will be quite an expensive operation.
+        """
+        pass
+
+    def _split_decomposed_graphs(self, graph_dict: dict) -> dict:
+        """
+        Build an indices dict to store the groups of atoms in each molecule.
+
+        Parameters
+        ----------
+        graph_dict : dict
+                Dict of decomposed graphs to be converted into correct particle species
+                indices.
+        Returns
+        -------
+        group_dict : dict
+                A dictionary of atoms and indices that specify that indices of
+                this species is in a molecule.
+        """
+        particle_groups = {}
+        for item in graph_dict:
+            indices_dict = {}
+            lengths = []
+            for i, particle_species in enumerate(self.species):
+                length = self.experiment.species[particle_species].n_particles
+                if i == 0:
+                    lengths.append(length)
+                else:
+                    lengths.append(length + lengths[i - 1])
+
+            for i, particle_species in enumerate(self.species):
+                if i == 0:
+                    indices_dict[particle_species] = np.sort(
+                        np.array(list(filter(lambda x: x < lengths[i], graph_dict[item])))
+                    ).tolist()
+                else:
+                    greater_array = list(
+                        filter(lambda x: x >= lengths[i - 1], graph_dict[item])
+                    )
+                    constrained_array = list(
+                        filter(lambda x: x < lengths[i], greater_array)
+                    )
+                    indices_dict[particle_species] = np.sort(
+                        np.array(constrained_array) - (lengths[i - 1])
+                    ).tolist()
+
+            particle_groups[item] = indices_dict
+
+        return particle_groups
+
+
+def build_smiles_graph(smiles_string: str) -> tuple:
+    """
+    Build molecular graphs from SMILES strings.
+
+    Parameters
+    ----------
+    smiles_string : str
+            SMILES string to use in the graph construction.
+
+    Returns
+    -------
+    smiles_graph :
+            Graph object returned by PySmiles
+    species : dict
+            A dict object containing species information about the molecule.
+    """
+    mol = read_smiles(smiles_string, explicit_hydrogen=True)
+    data = mol.nodes
+    species = {}
+    for i in range(len(data)):
+        item = data[i].get("element")
+        if item in species:
+            species[item] += 1
+        else:
+            species[item] = 0
+            species[item] += 1
```

Comment:
`species[item] = 1`

Reply:
it is cumulative, not set to 1

Reply:
You set it to zero and them add one. Am I missing something?

Reply:
resolved in b04555b